### ISSUE TITLE
no-descending-specificity error across components

### ISSUE THREAD

**Issue by danbriggs5** [styled-components/stylelint-processor-styled-components#201]
I'm getting a stylelint error: `no-descending-specificity`.

Given the simplified code:
```javascript
const Paragraph1 = styled.p`
	color: #000;

	> a {
		color: #00f;

		&:hover {
			color: #00e;
		}
	}
`;

const Paragraph2 = styled.p`
	color: #000;

	> a {
		color: #00f;

		&:hover {
			color: #00e;
		}
	}
`;
```

The error:
` 18:2  ✖  Expected selector ".selector2 > a" to come before selector ".selector1 > a:hover"   no-descending-specificity`

It is the `> a` selector in the second component that gets the error.

It appears that the `&:hover` from the first component is effecting the lint on the second component. If I remove the `&:hover` selector from the first component I no longer get the error.

Here is my .stylelintrc.js file for reference:
```javascript
module.exports = {
	processors: [
		'stylelint-processor-styled-components',
	],
	extends: [
		'stylelint-config-recommended',
		'stylelint-config-styled-components',
	],
	syntax: 'scss',
	rules: {
		indentation: 'tab',
	},
};

```

Please let me know if I'm missing something or if I can provide any more info. Happy to help :) For now I've just disabled the rule.

**Comment by danbriggs5** [comment]
@chinesedfan Thanks for finding that. It looks like the same error to me. I should have run a test with just stylelint before posting here. Thanks again!

**Comment by trainoasis** [comment]
Anyone fixed this somewhere already? Many threads about this but no solution so far? :)

**Comment by dawidk92** [comment]
Any updates about it?

**Comment by nosilleg** [comment]
@dawidk92 This is not a bug.  The linter is correctly telling you that `a:hover` has a higher specificity than `a`.

In your code you plan to never make `.selector2 > a` and `.selector1 > a` be the same `a`.  However, there is no way for the linter to know this, and since they _could_ be the same `a`, you get an error.

You could try `.selector2:not(.selector1) > a` and I'm guessing that would work, but it's not nice or scalable.

**Comment by dawidk92** [comment]
@nosilleg thank you for your response!
That's a shame that the linter doesn't know that these are two different components.

Besides that, I think that It's a much better solution to use `/* stylelint-disable-next-line no-descending-specificity */` instead.

**Comment by nosilleg** [comment]
@dawidk92 I agree about disabling the rule for most of the times I've seen this reported.

Another thought, are you willing/able to put a globally unique class (per component) on every child `a` of that component.  That would also solve your issue.

e.g.
```js
const Paragraph1 = styled.p`
	color: #000;

	> .a-p1 {
		color: #00f;

		&:hover {
			color: #00e;
		}
	}
`;

const Paragraph2 = styled.p`
	color: #000;

	> .a-p2 {
		color: #00f;

		&:hover {
			color: #00e;
		}
	}
`;
```
